### PR TITLE
onedrive: 2.3.13 ->! 2.4.2

### PR DESCRIPTION
**Pull request by SRGOM**
###### Motivation for this change
Version bump

Supersededs:  https://github.com/NixOS/nixpkgs/pull/83280 and https://github.com/NixOS/nixpkgs/pull/86707

New PR because I use shallow nixpkgs and github does't allow me to push on the old branch
###### Things done
 

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @fadenb

### CODE REVIEW

**Review by fadenb — APPROVED** [review]

**Comment by SRGOM** [comment]
Updated to 2.4.2, also started a new convention, as suggested by @danbst here- https://github.com/NixOS/nixpkgs/pull/83280#issuecomment-603337347 where if user intervention is required then the -> is followed by a bang. 

Please review and approve.

**Review by fadenb — APPROVED** [review]

**Comment by fadenb** [comment]
Is there anything else that needs to be done before this can be merged?

**Comment by Infinisil** [comment]
@Lassulus https://github.com/NixOS/nixpkgs/pull/89198#issuecomment-639269157

File: pkgs/applications/networking/sync/onedrive/default.nix
Comment (Infinisil) [inline pkgs/applications/networking/sync/onedrive/default.nix:24]: Why remove the maintainers?

**Comment by SuperSandro2000** [comment]
Version on master is already never.

**Pull request closed without merging**
